src/components/Comments/CommentForm.js

```diff
@@ -1,25 +1,77 @@
-import React, { PropTypes } from 'react';
+import React, { Component, PropTypes } from 'react';
 import { Input } from 'antd';
 import Avatar from '../Avatar';
 import './CommentForm.less';
 
-const CommentForm = ({ username, isSmall }) => (
-  <div className="CommentForm">
-    <Avatar username={username} size={(!isSmall) ? 40 : 32} />
-    <div className="CommentForm__text">
-      <Input type="textarea" placeholder="Write a comment" autosize={{ minRows: 2, maxRows: 6 }} />
-      <button className="CommentForm__text__button">Comment</button>
-    </div>
-  </div>);
+class CommentForm extends Component {
+
+  state = {
+    inputValue: this.props.inputValue,
+    isDisabledSubmit: false,
+  }
+
+  handleCommentTextChange = (e) => {
+    this.setState({ inputValue: e.target.value });
+  }
+
+  handleSubmit = (e) => {
+    e.stopPropagation();
+    this.setState({ isDisabledSubmit: true });
+    if (this.state.inputValue) {
+      this.props.onSubmit(this.props.parentPost, this.state.inputValue);
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
I think we should put lifecycle methods (except `render`) above our own methods.